### ISSUE TITLE
Keyboard closes randomly

### ISSUE THREAD

**Issue by sjiawjbssj** [brave/brave-browser#40076]
### Description

While entering a url the keyboard sometimes closes.

### Steps to reproduce

1. Try to enter a url

### Actual result

Keyboard closes even though I'm still typing.

### Expected result

Keyboard should stay open

### Reproduces how often

Easily reproduced

### Brave version

Brave 1.68.128, Chromium 127.0.6533.73

### Device

- Brand/model: Pixel 7
- Android version: Android 14

### Channel information

- [X] release (stable)
- [ ] beta
- [ ] nightly

### Reproducibility

- [ ] with Brave Shields disabled
- [ ] with Brave Rewards disabled
- [ ] in the latest version of Chrome

**Comment by ronaldjonnes** [comment]
I have experienced this issue for well over a year (maybe 2) and was ever hopeful that the developers would address it. Someone raised this in 2019 but nothing came of it.

**Comment by yonoff** [comment]
After swipig Brave from recent apps, when I open Brave and try to enter an address, the keyboard immediately closes, but if I try typing again, everything works fine.
The same issue occurs when using the Gboard and HeliBoard keyboards. There are no problems with keyboards in other applications.

Brave version 1.85.117
Pixel 7a - a16 (BP3A.251105.015)

https://github.com/user-attachments/assets/ef25ad2a-5c7b-4d13-a1e1-779ec397d596

**Comment by sjiawjbssj** [comment]
> After swipig Brave from recent apps, when I open Brave and try to enter an address, the keyboard immediately closes, but if I try typing again, everything works fine. The same issue occurs when using the Gboard and HeliBoard keyboards. There are no problems with keyboards in other applications.
> 
> Brave version 1.85.117 Pixel 7a - a16 (BP3A.251105.015)
> 
>  VID-20251227-WA0007.mp4

Do you have sync enabled? For me it stopped happening after disabling syncing.

**Comment by yonoff** [comment]
> Do you have sync enabled? For me it stopped happening after disabling syncing.

Yes, sync is enabled; I don't see the point in using a browser without it.

**Comment by bsclifton** [comment]
cc: @simoarpe @Olipro @SergeyZhukovsky

**Comment by SergeyZhukovsky** [comment]
@yonoff could you try to remove the bookmarks page from your Home Page? Does it help?

**Comment by simoarpe** [comment]
Is this issue reproducible also on Chrome/Chromium for Android? 
It seems upstream suffers a similar issue: https://issues.chromium.org/issues/40873378

**Comment by Olipro** [comment]
Since I don't think any of us can reproduce this - does anyone think it would be worth guiding someone through doing a logcat capture to see if anything relevant is showing up in there (on the off chance some sort of error or exception is causing it)

**Comment by yonoff** [comment]
> [@yonoff](https://github.com/yonoff) could you try to remove the bookmarks page from your Home Page? Does it help?

@SergeyZhukovsky 
Looks like it helps.
I set it up as shown in the screenshot, and so far the keyboard is working fine.
However, I would like it to work with bookmarks as well.

<img width="864" height="798" alt="Image" src="https://github.com/user-attachments/assets/f443834e-ab12-47e7-8e94-18b3ebe94d51" />

**Comment by sjiawjbssj** [comment]
> Since I don't think any of us can reproduce this - does anyone think it would be worth guiding someone through doing a logcat capture to see if anything relevant is showing up in there (on the off chance some sort of error or exception is causing it)

Could you try setting the bookmarks as your homepage and enable syncing? When I had that issue I was able to reproduce it on a second device. You might have to 
open the app -> open search -> close the app 
a couple of times until it happens.
You don't have to completely close the app, just going back to the home screen is enough.
